deepchecks/utils/single_sample_metrics.py

```diff
@@ -10,13 +10,60 @@
 #
 """Common metrics to calculate performance on single samples."""
 import numpy as np
+from sklearn.preprocessing import LabelBinarizer
 
 
-def per_sample_binary_cross_entropy(y_true: np.array, y_pred: np.array):
-    """Calculate binary cross entropy on a single sample."""
+def per_sample_cross_entropy(y_true: np.array, y_pred: np.array, eps=1e-15):
+    """Calculate cross entropy on a single sample.
+
+    This code is based on the code for sklearn log_loss metric, without the averaging over all samples. Licence below:
+    BSD 3-Clause License
+
+    Copyright (c) 2007-2021 The scikit-learn developers.
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright notice, this
+      list of conditions and the following disclaimer.
+
+    * Redistributions in binary form must reproduce the above copyright notice,
+      this list of conditions and the following disclaimer in the documentation
+      and/or other materials provided with the distribution.
+
+    * Neither the name of the copyright holder nor the names of its
+      contributors may be used to endorse or promote products derived from
+      this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+    FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+    DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+    SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+    OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+    """
     y_true = np.array(y_true)
-    return - (np.tile(y_true.reshape((-1, 1)), (1, y_pred.shape[1])) *
-              np.log(y_pred + np.finfo(float).eps)).sum(axis=1)
+
+    # Make y_true into one-hot
+    lb = LabelBinarizer()
+    lb.fit(list(range(y_pred.shape[1])))  # We assume that y_true is a list of corresponding indices in y_pred
+    transformed_labels = lb.transform(y_true)
+
+    if transformed_labels.shape[1] == 1:
+        transformed_labels = np.append(
+            1 - transformed_labels, transformed_labels, axis=1
+        )
+
+    # clip and renormalization y_pred
+    y_pred = np.clip(y_pred, eps, 1 - eps)
```

Comment:
```suggestion
    y_pred = y_pred.clip(eps, 1 - eps)
```
Just because prettier